### ISSUE TITLE
Implement HEALTHCHECK command

### ISSUE THREAD

**Issue by priyawadhwa** [GoogleContainerTools/kaniko#17]
See #1 for details

**Comment by chrisz100** [comment]
Healthcheck requires a different config struct:
https://github.com/moby/moby/blob/master/api/types/container/config.go

It seems to be not included in the open container specs - thus implementing this would require completely replacing the base for configs. 

After that - if I get the code right - it would simply mean passing the already prepared HealthConfig struct to the config to persist it. Which shouldn't be hard at all - but that requires HealthCheck to be existent in the container config. 

Ideas?

**Comment by chrisz100** [comment]
@dlorenc @priyawadhwa any idea?

**Comment by priyawadhwa** [comment]
Hey, sorry for the late response! We use the [go-containerregistry](https://github.com/google/go-containerregistry) library for working with our images, and right now the [config](https://github.com/google/go-containerregistry/blob/master/v1/config.go#L75) doesn't have a field for Healthcheck. You'd probably have to create the healthcheck config struct in go-containerregistry and add it to the config before starting to implement this command in kaniko.

**Comment by chrisz100** [comment]
So you suggest contributing to go-containerregistry and using the merged version for proceeding here, correct?

**Comment by dlorenc** [comment]
@chrisz100 correct! I can help you get that PR merged into go-containerregistry as well :)
